src/config-array.js

```diff
@@ -705,8 +729,20 @@ export class ConfigArray extends Array {
 		this.forEach((config, index) => {
 
 			if (!config.files) {
-				debug(`Anonymous universal config found for ${filePath}`);
-				matchingConfigIndices.push(index);
+
+				if (!config.ignores) {
+					debug(`Anonymous universal config found for ${filePath}`);
+					matchingConfigIndices.push(index);
+					return;
+				}
+
+				if (pathMatchesIgnores(filePath, this.basePath, config)) {
+					debug(`Matching config found for ${filePath} (based on ignores: ${config.ignores})`);
+					matchingConfigIndices.push(index);
+					matchFound = true;
```

Comment:
I think this config should still be treated as universal, i.e. I don't think this path should have `matchFound = true;`.

Reply:
Also, I think this misses `return;` so that it doesn't execute `debug(`Skipped config found for ${filePath} (based on ignores: ${config.ignores})`);`.